Tests/UnitTests/Scripts/tests.js

```diff
@@ -211,6 +211,241 @@ describe("clearTimeout", function () {
+        // => 'https://developer.mozilla.org/en-US/docs?foo=1&bar=2'
+        url.searchParams.set('foo', 999);
+        // href should change to reflect searchParams change
+        checkURL(url,
+            'https://developer.mozilla.org/en-US/docs?foo=999&bar=2', // href
+            'developer.mozilla.org',                                  // hostname
+            'https://developer.mozilla.org',                          // origin
+            '/en-US/docs',                                            // pathname
+            '?foo=999&bar=2'                                          // search
+        );
+    })
+
+    it("should update href after URLSearchParams are changed (Starting with 0 params)", function () {
+        // Augment URL with pathname and search
+        url = new URL(baseUrl + '/en-US/docs');
+        // => 'https://developer.mozilla.org/en-US/docs?foo=1&bar=2'
+
+        url.searchParams.set('foo', 999);
+
+        // href should change to reflect searchParams change
+        checkURL(url,
+            'https://developer.mozilla.org/en-US/docs?foo=999', // href
+            'developer.mozilla.org',                            // hostname
+            'https://developer.mozilla.org',                    // origin
+            '/en-US/docs',                                      // pathname
+            '?foo=999'                                          // search
+        );
+    })
+
+    // TODO: Does not identify and replace pre-established parts of a URL
+    //       ex. url already has '/en-US/docs' appended to it
+    // let url = new URL('/en-US/docs', url);
+    // => 'https://developer.mozilla.org/en-US/docs'
+
+    //new URL('/en-US/docs', "https://developer.mozilla.org/fr-FR/toto");
+    // => 'https://developer.mozilla.org/en-US/docs'
+
+
+    // TODO: Implement URL construction throws, currently does not check anything
+    // Invalid URLs:
+
+    //new URL('/en-US/docs', '');
+    // Raises a TypeError exception as '' is not a valid URL
+
+    //new URL('/en-US/docs');
+    // Raises a TypeError exception as '/en-US/docs' is not a valid URL
+
+
+    // Other cases:
+
+    //new URL('http://www.example.com',);
+    // => 'http://www.example.com/'
+
+    //new URL('http://www.example.com', B);
+    // => 'http://www.example.com/'
+
+    //new URL("", "https://example.com/?query=1");
+    // => 'https://example.com/?query=1' (Edge before 79 removes query arguments)
+
+    //new URL("/a", "https://example.com/?query=1");
+    // => 'https://example.com/a' (see relative URLs)
+
+    //new URL("//foo.com", "https://example.com");
+    // => 'https://foo.com' (see relative URLs)
+
+})
+
+// URLSearchParams
+describe("URLSearchParams", function () {
+
+    // -------------------------------- URLSearchParams Get --------------------------------
+
+    it("should retrieve null from empty searchParams", function () {
+        // Get Empty
+        let params = new URLSearchParams('');
```

Comment:
Prefer `const` over `let` where possible. Multiple places.

```suggestion
        const params = new URLSearchParams('');
```